packages/base/card-api.gts

```diff
@@ -2904,3 +2904,120 @@ type ElementType<T> = T extends (infer V)[] ? V : never;
 function makeRelativeURL(maybeURL: string, opts?: SerializeOpts): string {
   return opts?.maybeRelativeURL ? opts.maybeRelativeURL(maybeURL) : maybeURL;
 }
+
+type EmptySchema = {};
+
+type ArraySchema = {
+  type: 'array';
+  items: Schema;
+};
+
+type ObjectSchema = {
+  type: 'object';
+  properties: {
+    [fieldName: string]: Schema;
+  };
+};
+
+type DateSchema = {
+  type: 'string';
+  format: 'date' | 'date-time';
+};
+
+type PrimitiveSchema = {
+  type: 'string' | 'number' | 'boolean';
+};
+
+type Schema =
+  | EmptySchema
+  | ArraySchema
+  | ObjectSchema
+  | DateSchema
+  | PrimitiveSchema;
+
+/**
+ * Generates the JSON schema required for patching
+ *
+ * @param def - The BaseDef to generate the patch call specification for.
+ * @param typeChain - A list of the types
+ */
+function _generatePatchCallSpecification(
+  def: typeof BaseDef,
+  typeChain: string[],
+) {
+  // If we have already seen this type, then we are in a recursive loop
+  // and we should stop. This can happen when a card has a field that
+  // references itself.
+  if (typeChain.includes(def.name)) {
+    return undefined;
+  } else {
+    // This must be a copy, otherwise we're mutating an array
+    // used through the recursive calls
+    typeChain = [...typeChain, def.name];
```

Comment:
so contained fields cannot recurse--which are fields that appear in the `attributes` of the card instance doc. Linked fields can recurse--however, i think you are only building JSON schema for the `attributes` section of the card document, correct?